contracts/ssnlist.scilla

```diff
@@ -810,16 +840,16 @@ transition WithdrawStakeRewards(ssn_operator : ByStr20, deleg : ByStr20)
   IsPaused;
   IsProxy;
   DelegExists ssn_operator deleg;
-  DistributeStakeRewards deleg ssn_operator
+  WithdrawalStakeRewards deleg ssn_operator
 end
 (* @dev: Withdraw all amount owed by the contract. Used by delegs. *)
 (* @param ssn: Address of the SSN to which stake had been delegated. *)
 (* @param initiator: The original caller who called the proxy. *)
-(* @note: Withdraws stake + buffer + reward in one go. *)
-transition WithdrawStakeAmt(ssn: ByStr20, initiator : ByStr20)
+(* @param amt: The amount caller who wanted to withdraw from *)
+transition WithdrawStakeAmt(ssn: ByStr20, initiator : ByStr20, amt: Uint128)
   IsPaused;
   IsProxy;
-  DistributeStakeAmt initiator ssn
+  WithdrawalStakeAmt initiator ssn amt
```

Comment:
should we add   DistributeStakeRewards deleg ssn_operator

Reply:
It is feasible theoretically, but my concern is, in the future, we might need to cut withdraw rewards into multiple transactions. As we discussed offline, if one has 100 cycles to withdraw, we will need him claim 2 or 3 times in case the gasfee would hit some limit. So here, if we want to send his delegate amount and rewards in one go, things will be a bit tough due to the same issue (I mean the gasfee issue).

Reply:
ok